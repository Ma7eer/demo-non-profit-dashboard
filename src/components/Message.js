import { toast } from "react-toastify";

const settings = {
  position: "top-right",
  autoClose: 4000,
  hideProgressBar: true,
  closeOnClick: true,
  pauseOnHover: true,
  draggable: false
};

export default (color, text) =>
  color === "error"
    ? toast.error(text, settings)
    : color === "warn"
    ? toast.warn(text, settings)
    : toast.success(text, settings);
