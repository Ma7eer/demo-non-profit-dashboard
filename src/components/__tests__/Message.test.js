import React from 'react'
import { toast } from 'react-toastify'
import renderer from 'react-test-renderer'
import { ToastContainer } from 'react-toastify'
import Message from '../Message'

const settings = {
  position: 'top-right',
  autoClose: 4000,
  hideProgressBar: true,
  closeOnClick: true,
  pauseOnHover: true,
  draggable: false
}

test('it returns a toastr message', () => {
  let text = 'my Error message'
  let component = renderer.create(<ToastContainer />)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
  //   expect(Message('error', text)).toBe(toast.error(text, settings))
})
