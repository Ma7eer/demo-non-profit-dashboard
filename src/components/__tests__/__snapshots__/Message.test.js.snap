// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it returns a toastr message 1`] = `
<div
  className="Toastify"
/>
`;
